.ci/bumpStackVersion.groovy

```diff
@@ -0,0 +1,126 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import groovy.transform.Field
+
+@Library('apm@current') _
+
+// To store all the latest snapshot versions
+@Field def latestVersions
+
+pipeline {
+  agent { label 'linux && immutable' }
+  environment {
+    REPO = 'observability-dev'
+    HOME = "${env.WORKSPACE}"
+    NOTIFY_TO = credentials('notify-to')
+    PIPELINE_LOG_LEVEL='INFO'
+  }
+  options {
+    timeout(time: 1, unit: 'HOURS')
+    buildDiscarder(logRotator(numToKeepStr: '20', artifactNumToKeepStr: '20'))
+    timestamps()
+    ansiColor('xterm')
+    disableResume()
+    durabilityHint('PERFORMANCE_OPTIMIZED')
+    rateLimitBuilds(throttle: [count: 60, durationName: 'hour', userBoost: true])
+    quietPeriod(10)
+  }
+  parameters {
+    booleanParam(name: 'DRY_RUN_MODE', defaultValue: false, description: 'If true, allows to execute this pipeline in dry run mode, without sending a PR.')
+  }
+  stages {
+    stage('Checkout') {
+      steps {
+        git(credentialsId: '2a9602aa-ab9f-4e52-baf3-b71ca88469c7-UserAndToken', url: "https://github.com/elastic/${REPO}.git")
+      }
+    }
+    stage('Fetch latest versions') {
+      steps {
+        script {
+          latestVersions = artifactsApi(action: 'latest-versions')
+        }
+      }
+    }
+    stage('Send Pull Request'){
+      options {
+        warnError('Pull Requests failed')
+      }
+      steps {
+        generateSteps()
+      }
+    }
+  }
+  post {
+    cleanup {
+      notifyBuildResult()
+    }
+  }
+}
+
+def generateSteps(Map args = [:]) {
+  def projects = readYaml(file: '.ci/.bump-stack-version.yml')
+  def parallelTasks = [:]
+  projects['projects'].each { p ->
```

Comment:
an alternative implementation

```
matrix(
          agent: 'ubuntu-20',
          axes: [
            axis('JOB_PROJECT', projects ),
            axis('JOB_BRANCHES', branches )
          ]
        ){
        ...
        }
```

Reply:
Cool, I'll give a go!

Reply:
matrix does not work well with `Field` variables :/ 

```
10:22:33  ERROR: Pull Requests failed
10:22:33  Also:   groovy.lang.MissingPropertyException: No such property: latestVersions for class: groovy.lang.Binding
10:22:33  Also:   groovy.lang.MissingPropertyException: No such property: latestVersions for class: groovy.lang.Binding
10:22:33  groovy.lang.MissingPropertyException: No such property: latestVersions for class: groovy.lang.Binding
10:22:33  	at groovy.lang.Binding.getVariable(Binding.java:63)
10:22:33  	at org.jenkinsci.plugins.scriptsecurity.sandbox.groovy.SandboxInterceptor.onGetProperty(SandboxInterceptor.java:270)
10:22:33  	at org.kohsuke.groovy.sandbox.impl.Checker$7.call(Checker.java:353)
10:22:33  	at org.kohsuke.groovy.sandbox.impl.Checker.checkedGetProperty(Checker.java:357)
10:22:33  	at org.kohsuke.groovy.sandbox.impl.Checker.checkedGetProperty(Checker.java:333)
10:22:33  	at org.kohsuke.groovy.sandbox.impl.Checker.checkedGetProperty(Checker.java:333)
10:22:33  	at com.cloudbees.groovy.cps.sandbox.SandboxInvoker.getProperty(SandboxInvoker.java:29)
10:22:33  	at com.cloudbees.groovy.cps.impl.PropertyAccessBlock.rawGet(PropertyAccessBlock.java:20)
10:22:33  	at WorkflowScript.generateSteps(WorkflowScript:60)
10:22:33  	at matrix.call(matrix.groovy:35)
```

So I'll not use it for the time being (9eacd58 is the commit with the proposal)

Reply:
I said nothing! I had a wrong variable name :/

![image](https://user-images.githubusercontent.com/2871786/115532808-09e52500-a28e-11eb-85f6-0466f503990c.png)